### PR TITLE
Add namespace scope to compute template operations

### PR DESCRIPTION
**Pull request by daikeshi**
## Why are these changes needed?
This PR adds the namespace scope to get/list/delete for compute templates.  
* Added a `namespace` field to compute template
* Added a `namespace` field to compute template get/list/delete request
* Added a `namespace` option to cli for compute templates 
* If namespace is not provided, it will use `ray-system` as the default 

*** Update 1 ***
To follow the decisions made in https://github.com/ray-project/kuberay/pull/237,  following changes are also included in this PR:
* Make namespace a required parameter for compute template operations
* Bump corresponding apiserver API version to v1alpha2
* Updated the doc  

*** Update 2 ***
* Add an endpoint to list all compute templates in all namespaces

## Related issue number
Closes https://github.com/ray-project/kuberay/issues/238

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by daikeshi** [comment]
@Jeffwan This PR is ready for review now. It's updated based on the changes I did in https://github.com/ray-project/kuberay/pull/237

File: proto/config.proto [hunk 1]
```diff
@@ -151,30 +172,41 @@ message ListImageTemplatesResponse {
   repeated ImageTemplate image_templates = 1;
 }
 
+message ListAllImageTemplatesRequest {
```
Comment (daikeshi) [inline proto/config.proto:175]: I also made the changes to the Image template, but let me know if this is not necessary.

Comment (Jeffwan) [reply proto/config.proto:175]: This should be ok

**Comment by Jeffwan** [comment]
The downside is we may introduce `namespace` (kubernetes domain concept) to generic configs like compute templates which should be simple and clean. Seems we don't have other ways at this moment. Once we supports DB, I think these values can be moved outside kubernetes. 

The other thing is whether we should use different folder for v1alpha1 and v1alpha2 for PB definitions? This is out of scope of this story and we can have some follow up discussion later.

File: proto/config.proto (hunk 1 above)
Comment (Jeffwan) [reply proto/config.proto:175]: I think this is ok

**Comment by Jeffwan** [comment]
Thanks for the contribution. Everything looks good to me and let's merge the change

**Comment by daikeshi** [comment]
> The downside is we may introduce `namespace` (kubernetes domain concept) to generic configs like compute templates which should be simple and clean. Seems we don't have other ways at this moment. Once we supports DB, I think these values can be moved outside kubernetes.

I see your points. Yeah, I agree. We can address this when we start adopting DB to store configs.  

> The other thing is whether we should use different folder for v1alpha1 and v1alpha2 for PB definitions? This is out of scope of this story and we can have some follow up discussion later.

That's a good point. I guess it depends on if we would like to support multiple API versions at the same time. We can discuss more on this.

**Pull request merged**